### PR TITLE
Move replicas to KubernetesResource from eventlistener spec

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes
`replicas` field will be useful incase of `KubernetesResource` resource not `CustomResource` so keeping `replicas` at eventlistener spec level creating confusion while creating `CustomResource` object.
In order to avoid such confusions moving `replicas` field  from [eventlistener spec](https://github.com/tektoncd/triggers/blob/main/pkg/apis/triggers/v1alpha1/event_listener_types.go#L62) to [KubernetesResource](https://github.com/tektoncd/triggers/blob/main/pkg/apis/triggers/v1alpha1/event_listener_types.go#L79). Deprecating `replicas` from eventlistener spec and will be removed in future.

Fixes: https://github.com/tektoncd/triggers/issues/1019

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Deprecated replicas from EventListener spec level and added replicas to KubernetesResource.
```

### CODE REVIEW

File: docs/eventlisteners.md [hunk 1]
```diff
@@ -263,6 +261,13 @@ spec:
 
 With the help of `kubernetesResource` user can specify [PodTemplateSpec](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3704).
 
+##### Replicas
+
+The `replicas` field is optional. By default, the number of replicas of EventListener is 1.
+If you want to deploy more than one pod, you can specify the number to `replicas` field.
```
Comment (mpetason) [inline docs/eventlisteners.md:267]:
```suggestion
If you want to deploy more than one pod, you can specify the amount with the `replicas` field.
```

File: docs/eventlisteners.md [hunk 2]
```diff
@@ -263,6 +261,13 @@ spec:
 
 With the help of `kubernetesResource` user can specify [PodTemplateSpec](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3704).
 
+##### Replicas
+
+The `replicas` field is optional. By default, the number of replicas of EventListener is 1.
```
Comment (mpetason) [inline docs/eventlisteners.md:266]:
```suggestion
The `replicas` field is optional. By default, the number of replicas for the EventListener is 1.
```

File: docs/eventlisteners.md [hunk 3]
```diff
@@ -263,6 +261,13 @@ spec:
...
+If you want to deploy more than one pod, you can specify the number to `replicas` field.
+
+**Note:** If user sets `replicas` field while creating/updating eventlistener yaml then it won't respects replicas values edited by user manually on deployment or through any other mechanism (ex: HPA).
```
Comment (mpetason) [inline docs/eventlisteners.md:269]:
```suggestion
**Note:** If the user sets the `replicas` field while creating/updating the eventlistener yaml then it won't respect the replicas value edited by the user manually on deployment or through any other mechanism (ex: HPA).
```

**Review by mpetason — COMMENTED** [review]
small edits that may read better.

File: docs/eventlisteners.md (hunk 1 above)
Comment (savitaashture) [reply docs/eventlisteners.md]: Updated

**Comment by savitaashture** [comment]
@dibyom @mpetason Thank you for reviewing
Addressed review comment

**Review by dibyom — APPROVED** [review]
/lgtm

**Pull request merged**